source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -0,0 +1,372 @@
+  Stats::Scope& cluster_stats_scope = cluster_info->statsScope();
+
+  // Combining timeouts+retries - retries are counted  as separate requests
+  // (alternative: each request including the retries counted as 1).
+  uint64_t timeouts = cluster_stats.upstream_rq_timeout_.value() +
+                      cluster_stats.upstream_rq_per_try_timeout_.value();
+
+  pushNewValue(cluster_stats_cache.timeouts_, timeouts);
+
+  // Combining errors+retry errors - retries are counted as separate requests
+  // (alternative: each request including the retries counted as 1)
+  // since timeouts are 504 (or 408), deduce them from here ("-" sign).
+  // Timeout retries were not counted here anyway.
+  uint64_t errors = cluster_stats_scope.counter("upstream_rq_5xx").value() +
+                    cluster_stats_scope.counter("retry.upstream_rq_5xx").value() +
+                    cluster_stats_scope.counter("upstream_rq_4xx").value() +
+                    cluster_stats_scope.counter("retry.upstream_rq_4xx").value() -
+                    cluster_stats.upstream_rq_timeout_.value();
+
+  pushNewValue(cluster_stats_cache.errors_, errors);
+
+  uint64_t success = cluster_stats_scope.counter("upstream_rq_2xx").value();
+  pushNewValue(cluster_stats_cache.success_, success);
+
+  uint64_t rejected = cluster_stats.upstream_rq_pending_overflow_.value();
+  pushNewValue(cluster_stats_cache.rejected_, rejected);
+
+  // should not take from upstream_rq_total since it is updated before its components,
+  // leading to wrong results such as error percentage higher than 100%
+  uint64_t total = errors + timeouts + success + rejected;
+  pushNewValue(cluster_stats_cache.total_, total);
+
+  ENVOY_LOG(trace, "{}", printRollingWindows());
+}
+
+void HystrixSink::resetRollingWindow() { cluster_stats_cache_map_.clear(); }
+
+void HystrixSink::addStringToStream(absl::string_view key, absl::string_view value,
+                                    std::stringstream& info) {
+  std::string quoted_value = absl::StrCat("\"", value, "\"");
+  addInfoToStream(key, quoted_value, info);
+}
+
+void HystrixSink::addIntToStream(absl::string_view key, uint64_t value, std::stringstream& info) {
+  addInfoToStream(key, std::to_string(value), info);
+}
+
+void HystrixSink::addInfoToStream(absl::string_view key, absl::string_view value,
+                                  std::stringstream& info) {
+  if (!info.str().empty()) {
+    info << ", ";
+  }
+  std::string added_info = absl::StrCat("\"", key, "\": ", value);
+  info << added_info;
+}
+
+void HystrixSink::addHystrixCommand(ClusterStatsCache& cluster_stats_cache,
+                                    absl::string_view cluster_name,
+                                    uint64_t max_concurrent_requests, uint64_t reporting_hosts,
+                                    std::chrono::milliseconds rolling_window_ms,
+                                    std::stringstream& ss) {
+
+  std::stringstream cluster_info;
```

Comment:
Why create this variable?  Why not just append everything to `ss` directly?  (Same comment for the other places where this is done).